### DISCUSSION TITLE
Importing and converting existing Kubernetes YAML into Pkl

### DISCUSSION THREAD

**Discussion by luistrigueiros** [apple/pkl#52] [Q&A]
Is there a way to quick start from existing Kubernetes YAML and using some tool convert it into Pkl ?
Once is in Pkl start using the Pkl feature to enhance the process of generating this YAML.

**Comment by codebendr-product-designer** [comment]
Good question, I have been thinking similar but for JSON exported from Figma where I can import and extract design tokens for platform specific usage. 🤔

**Reply by holzensp** [reply]
Many schemas in [`pkl-pantry`](https://github.com/apple/pkl-pantry) and elsewhere come with a `convert.pkl` script. The K8s schemas are a somewhat special in that they're _generated_ from K8s's OpenAPI spec ([see `pkl-k8s`](https://github.com/apple/pkl-k8s)), so the `convert.pkl` script is published [in `k8s.contrib` here on `pkl-pantry`](https://pkl-lang.org/package-docs/pkg.pkl-lang.org/pkl-pantry/k8s.contrib/current/convert/index.html).

**Reply by luistrigueiros** [reply]
Yes I am see it now, but how can I use this converter ?

**Reply by jasongwartz** [reply]
@luistrigueiros There's a usage example in the source: https://github.com/apple/pkl-pantry/blob/k8s.contrib@1.0.0/packages/k8s.contrib/convert.pkl#L29

**Reply by luistrigueiros** [reply]
Hi @jasongwartz 

I am trying to run the command above but I am getting the error "The system cannot find the file specified."
I am running it in windows at the moment the command line I have build is:

---
java -jar jpkl.jar eval -p input=nginx-deployment.yaml -o deployment.pkl package://pkl.pkl-lang.org/pkl-pantry/k8s.contrib@<version>#/convert.pkl
---

![screenshot 4](https://github.com/apple/pkl/assets/590567/39f42aa8-33b6-4093-9dcb-acad7af8e3c0)

**Reply by jasongwartz** [reply]
@luistrigueiros I think you need to replace the `<version>` with an actual version. For example, on [the convert page on the package index](https://pkl-lang.org/package-docs/pkg.pkl-lang.org/pkl-pantry/k8s.contrib/current/convert/index.html), there's a `Module URI` specified (with a copy button), that (as of this writing) is `package://pkg.pkl-lang.org/pkl-pantry/k8s.contrib@1.0.0#/convert.pkl`.

I don't have a Windows machine to test, but I just ran `pkl eval -p input=deployment.yml -o deployment.pkl package://pkg.pkl-lang.org/pkl-pantry/k8s.contrib@1.0.0#/convert.pkl` (on my local Mac) and it worked as expected.

**Reply by luistrigueiros** [reply]
Hi @jasongwartz 

Oh ok, thank you first. 
I believe we have bug on window

---

C:\tools\pkl-lang.org\pkl-tutorial>java -jar jpkl.jar eval -p input=nginx-deployment.yaml -o deployment.pkl package://pkg.pkl-lang.org/pkl-pantry/k8s.contrib@1.0.0#/convert.pkl
An unexpected error has occurred. Would you mind filing a bug report?
---


![screenshot 5](https://github.com/apple/pkl/assets/590567/cb50898f-d086-4ac3-add3-7cf1bcc9e5d3)

**Reply by luistrigueiros** [reply]
Hi @jasongwartz 

I was able to do the conversion using linux alpine installation on a ubuntu over WSL

![screenshot 8](https://github.com/apple/pkl/assets/590567/02001548-3f80-43d0-a68f-8fd083e5ac3a)

**Reply by holzensp** [reply]
Yes, Windows support is on the TODO list, but not currently there, I'm afraid.

**Reply by luistrigueiros** [reply]
Thank you I have found in the Kubernets examples what I was looking for
